rakan/cRakan/src/Runner.h

```diff
@@ -24,68 +24,96 @@ class Runner {
  //////////////////////////////////////////////////////////////////////////////
  // Construction / Initialization
  //////////////////////////////////////////////////////////////////////////////
-  /*
-  * Default constructor.
+
+  /**
+  * Default constructor. Used ONLY for testing.
   */
   Runner();
 
+  /**
+   * Constructs a Runner instance that will run on a graph with num_precincts
+   * number of precincts and num_districts number of districts.
+   * 
+   * @param       num_precincts   the number of precincts on the running graph
+   * @param       num_districts   the number of districts on the running graph
+   */
   Runner(uint32_t num_precincts, uint32_t num_districts);
 
-  void add_node(uint32_t node_id, uint32_t county, uint32_t minority, uint32_t majority);
-
+  /**
+   * Destructor.
+   */
+  ~Runner();
+
+  /**
+   * Adds a new node to the graph this Runner instance runs on.
+   * 
+   * @param   node_id               the node id of the node to add
+   * @param   county                the county the to be added node resides in
+   * @param   majority_population   the majority population in the node
+   * @param   minority_population   the minority population in the node
+   */
+  void add_node(uint32_t node_id,
+                uint32_t county,
+                uint32_t majority_population,
+                uint32_t minority_population);
+
+  /**
+   * Adds an edge from node_one to node_two and from node_two to node_one
+   * on the graph this Runner runs on.
+   * 
+   * @param       node_one      the node to add an edge from node_two to
+   * @param       node_two      the node to add an edge from node_one to
+   */
   void add_edge(uint32_t node_one, uint32_t node_two);
 
-  // Returns a vector of vectors of uint32_t, where each vector
-  // represents the changes made to the graph in each step of 
-  // the walk.
-  vector<vector<uint32_t>&>& getMaps();
-  /*
-  * Sets the district assignments according to the given vector.
-  * Vector is interpreted as the node_id corresponding to the index
-  * of the district number inside of the vector.
-  * 
-  * @param    districts     a vector of uint32_t where each index of
-  *                         the vector corresponds to the node id.
+  /**
+  * Sets the district assignments according to the given vector. Vector is
+  * interpreted as the node_id corresponding to the index of the district
+  * number inside of the vector.
   * 
+  * @param    districts     a vector of uint32_t where each index of the vector
+  *                         corresponds to the node id
   */
   void set_districts(vector<uint32_t>& districts);
```

Comment:
Similar reasoning as above. Should this return true/false based on invalid/valid input?

Reply:
Sure